app/models/manageiq/providers/workflows/automation_manager/workflow_instance.rb

```diff
@@ -99,4 +102,21 @@ def run(zone: nil, role: "automation", object_type: nil, object_id: nil)
 
     run_queue(:zone => zone, :role => role, :object => object) if next_state.present?
   end
+
+  class << self
+    def floe_docker_runner
+      @floe_docker_runner ||= begin
```

Comment:
Super-minor, but the begin is redundant, since the if condition is effectively a single expression.

Reply:
I think alternatively you can put on the next line
```ruby
@floe_docker_runner ||=
  if MiqEnvironment::Command.is_podified?
    Floe::Workflow::Runner::Kubernetes.new
  elsif MiqEnvironment::Command.is_appliance?
    Floe::Workflow::Runner::Podman.new
  else
    Floe::Workflow::Runner::Docker.new
  end
```

or of course

@floe_docker_runner ||= new_runner_instance

def runner_class
  if MiqEnvironment::Command.is_podified?
    Floe::Workflow::Runner::Kubernetes.new
  elsif MiqEnvironment::Command.is_appliance?
    Floe::Workflow::Runner::Podman.new
  else
    Floe::Workflow::Runner::Docker.new
  end
```

or `runner_class.new` - which ever floats your boat.
probably nice to have the `new` hidden for future shenanigans. 

also
does it make sense to have
```ruby
class MiqEnvironment::Command
  def self.mode
    returns :kubernetes, :podman, :docker
  end
end
```

Reply:
Yeah the current code looks very different from this older diff.  I do like having something like a `container_runner_command` in `MiqEnvironment`